trusted-firmware-m/config/config_default.cmake

```diff
@@ -167,7 +167,8 @@ set(MBEDCRYPTO_PATH                     "DOWNLOAD"  CACHE PATH      "Path to Mbe
 set(MBEDCRYPTO_VERSION                  "mbedtls-2.25.0" CACHE STRING "The version of Mbed Crypto to use")
 set(MBEDCRYPTO_GIT_REMOTE               "https://github.com/ARMmbed/mbedtls.git" CACHE STRING "The URL (or path) to retrieve MbedTLS from.")
 set(MBEDCRYPTO_BUILD_TYPE               "${CMAKE_BUILD_TYPE}" CACHE STRING "Build type of Mbed Crypto library")
-set(TFM_MBEDCRYPTO_CONFIG_PATH          "${CMAKE_SOURCE_DIR}/lib/ext/mbedcrypto/mbedcrypto_config/tfm_mbedcrypto_config_default.h" CACHE STRING "Config to use for Mbed Crypto")
+set(TFM_MBEDCRYPTO_CONFIG_TYPE          "PATH" CACHE INTERNAL "Type of the *_MBEDCRYPTO_CONFIG_PATH variables")
```

Comment:
Are you sure `INTERNAL` at this location is correct ?
From the docs:
> INTERNAL
>
>   ..... Use of this type implies FORCE.

so I would think this would result in this line overruling our setting applied with `-DTFM_MBEDCRYPTO_CONFIG_TYPE=STRING`.

When I said `INTERNAL` I meant that other code could set this as INTERNAL to ensure its value, like:
```
set(TFM_MBEDCRYPTO_CONFIG_TYPE "STRING" CACHE INTERNAL "Config type in NCS is STRING")
.....
# Code that will later do:
include(trusted-firmware-m/config/config_default.cmake)
```
(as we use `-D` to set this value, then it doesn't really matter to us.)

But in this code INTERNAL was not intended.

Please correct me if this code actually work as you intended.

```suggestion
# Below is the cache type for TFM_MBEDCRYPTO_CONFIG_PATH.
# This is intended to be controlled by an external build system to support relative path.
# Relative paths can be supported with one of:
# - set(TFM_MBEDCRYPTO_CONFIG_TYPE "STRING" CACHE INTERNAL "Internal setting")
# - -DTFM_MBEDCRYPTO_CONFIG_TYPE=STRING
# before config_default.cmake is loaded.
set(TFM_MBEDCRYPTO_CONFIG_TYPE "PATH" CACHE STRING "Type of the *_MBEDCRYPTO_CONFIG_PATH variables, used internally")
```

Reply:
changed to `STRING` (y)